app/models/manageiq/providers/openstack/inventory/parser/network_manager.rb

```diff
@@ -186,7 +186,7 @@ def find_device_object(network_port)
       # TODO(slucidi): replace this query for hosts once the infra manager uses the graph inventory system
       host = collector.manager.hosts.where(:uid_ems => network_port.device_id).first
       return host || persister.vms.lazy_find(network_port.device_id)
-    when "network:router_interface", "network:router_ha_interface", "network:ha_router_replicated_interface"
+    when /^network\:.*?$/
```

Comment:
:+1: for removing the string list and replacing with a more generic selector, but I think it is needed look for _router_ word too, like ```/\Anetwork\:router/```.